bso/structural_design/element/flat_shell.cpp

```diff
@@ -359,6 +406,18 @@ namespace bso { namespace structural_design { namespace element {
 		return bso::utilities::geometry::quadrilateral::getArea() * mThickness;
 	} // getVolume()
 	
+	double flat_shell::getStressCenter(const double& alpha /* 0*/, const double& beta /* 1.0 / sqrt(3)*/) const // NOTE: only in-plane stresses are considered (dKQ stresses are ignored) because of the application in topology optimization
+	{
+		double sx, sy, sxy, I1, J2D, DPStress;
+		sx = mStress(0);
+		sy = mStress(1);
+		sxy = mStress(2);
+		I1 = sx + sy;
+		J2D = (1.0/3) * (pow(sx, 2) + pow(sy, 2) - sx * sy + 3 * pow(sxy, 2));
+		DPStress = (1.0 / beta) * (sqrt(J2D) + alpha * I1); // this value should not exceed 1
+		return DPStress;
+	} // getStressCenter() - NOTE: if alpha & beta are not inserted in the function call, the Von Mises stress is obtained
+
```

Comment:
Consider renaming this function to `getStressAtCenter`, it may be misleading otherwise